File: m/rls/rlb/txcop.c
/* file -  tx_cop - copy counted string */
#include "c:\rid\rider.h"
#include "c:\rid\txdef.h"
/* code -  tx_cop - copy counted */
char *tx_cop(
char *src ,
char *dst ,
int cnt )
{ if( cnt == 0)return ( dst );
  while ( cnt-- != 0) {
    if( (*dst = *src) == 0)break;
    ++dst, ++src;
  } 
  *dst = 0;
  return ( dst);
} 

File: m/rls/txcop.r
file	tx_cop - copy counted string
include rid:rider
include rid:txdef

code	tx_cop - copy counted

  func	tx_cop
	src : * char
	dst : * char
	cnt : int			; buffer length (excluding zero)
	()  : * char			; at terminator
  is	reply dst if cnt eq		; crazy
	while cnt-- ne			; got more
	  quit if (*dst = *src) eq	;
	  ++dst, ++src			;
	end				;
	*dst = 0			; force termination
	reply dst			;
  end
